pkg/auth/providers/oidc/oidc_client.go

```diff
@@ -4,37 +4,51 @@ import (
 	"context"
 	"crypto/tls"
 	"crypto/x509"
+	"fmt"
 	"net/http"
 
 	"github.com/coreos/go-oidc/v3/oidc"
 )
 
-func AddCertKeyToContext(ctx context.Context, certificate, key string) (context.Context, error) {
+func getClientCertificates(certificate, key string) ([]tls.Certificate, error) {
+	cert, err := tls.X509KeyPair([]byte(certificate), []byte(key))
+	if err != nil {
+		return nil, fmt.Errorf("could not parse cert/key pair: %w", err)
+	}
+
+	return []tls.Certificate{cert}, nil
+}
+
+func getHTTPClient(certificate, key string) (*http.Client, error) {
+	transport := http.DefaultTransport.(*http.Transport).Clone()
+	transport.TLSClientConfig = &tls.Config{
+		MinVersion: tls.VersionTLS12,
```

Comment:
Let's do this based on the existing [TLSMinVersion](https://github.com/rancher/rancher/blob/649110f64f1e462b4b979ca2ecb3eeea429c35a4/pkg/settings/setting.go#L106) setting, so that users can still set TLS 1.3 as the minimum, if they so wish.

Reply:
While I agree with what's proposed here, I'd be worried of introducing breaking changes. Should this be included in this PR or instead moved to another one for a release that would allow such a change?

Reply:
I reverted this since it wasn't explicitly set before, but don't mind to set it to the feature flag if that's what's desired.

Reply:
Thanks Ethan.

@pjbgf To add to this, although no longer needed for this PR, I'm not sure we'd want to use this settings at all since for now it's only used for the rancher server, not clients and [the docs mention](https://ranchermanager.docs.rancher.com/getting-started/installation-and-upgrade/installation-references/tls-settings) TLS Min setting for accepting connections.

If that's something that's wanted for outgoing connection then we'd need to see if we want to reuse that settings for it, or possibly introduce another setting.

Reply:
Fair point @tomleb, if we were to introduce that behaviour from a client perspective we probably should introduce a client version of that setting.